From: John Rigby <jrigby@freescale.com>
Date: Thu, 24 Aug 2006 12:56:38 -0600
Subject: [PATCH] nativendian audio

The original audio server uses little endian (LE) audio.
On bigendian (BE) target like ppc with a bigendian audio
codec this results in a conversion from BE to LE in
user space and then another convers from LE to BE
in kernel space in the sound driver.

This patch changes qsoundqss_qws to use native endian (NE)
eliminating the need to the extra conversions and also
eliminating the need for all clients of qsoundqss to
be patched to use LE.

Signed-off-by: John Rigby <jrigby@freescale.com>

---

 qt2/src/kernel/qsoundqss_qws.cpp |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

3780822ed791186ffe8a7d34da83d5885856bfd4
diff --git a/qt2/src/kernel/qsoundqss_qws.cpp b/qt2/src/kernel/qsoundqss_qws.cpp
index 23dff39..76fb169 100644
--- a/qt2/src/kernel/qsoundqss_qws.cpp
+++ b/qt2/src/kernel/qsoundqss_qws.cpp
@@ -1501,10 +1501,10 @@ #endif // QT_QSS_MANAGED_DSP
 	    if ( ioctl(fd, SNDCTL_DSP_SETFRAGMENT, &v) )
 		qWarning("Could not set fragments to %08x",v);
 #ifdef QT_QWS_SOUND_16BIT
-	    v=AFMT_S16_LE; if ( ioctl(fd, SNDCTL_DSP_SETFMT, &v) )
+	    v=AFMT_S16_NE; if ( ioctl(fd, SNDCTL_DSP_SETFMT, &v) )
 		qWarning("Could not set format %d",v);
-	    if ( AFMT_S16_LE != v )
-		qDebug("Want format %d got %d", AFMT_S16_LE, v);
+	    if ( AFMT_S16_NE != v )
+		qDebug("Want format %d got %d", AFMT_S16_NE, v);
 #else
 	    v=AFMT_U8; if ( ioctl(fd, SNDCTL_DSP_SETFMT, &v) )
 		qWarning("Could not set format %d",v);
-- 
1.3.3

